### ISSUE TITLE
How about ECS agents that fail to start completely?

### ISSUE THREAD

**Issue by scytacki** [silinternational/ecs-agent-monitor#5]
First, thanks for sharing this code.

Have you had problems with ECS agents that don't start when the autoscaler starts a new instance? It seems this could happen and then mess up any autoscaling policies that assume all instances are tracked by ECS.  I've never seen it happen but we've just got started with ECS.

If it is a problem the lamba function could be extended to know which autoscaler is managing the cluster, and then look for instances in the autoscaler that are not part of the cluster.

**Comment by fillup** [comment]
I have not seen that happen, although we haven't been auto-scaling our clusters, we just increase them as needed, but even then it has worked fine. 

And actually we developed this a while ago when the ECS agent wasn't very stable and was crashing on a pretty regular bases, like multiple times a week on at least one machine in our cluster. Amazon / ECS has gotten much better though at restarting the agent container and I haven't seen it go to a `false` state in a very long time (many months).

We added the tracking via Firebase because we found that the agent crashes quite regularly, and the lambda script would see it down and replace the instance, but the agent would have started back up on its own anyway and been fine. After activating this lambda it was replacing multiple instances per day and we had never seen the problem that frequently. Since we've been waiting to see it fail multiple times in a row before replacing it has not replaced a single instance. So I think the ECS Agent is quite stable now.
